.github/actions/build-frontend/action.yml

```diff
@@ -46,10 +46,24 @@ runs:
         yarn build:staging
       shell: bash
 
+    - name: SonarCloud Scan
+      uses: sonarsource/sonarcloud-github-action@de2e56b42aa84d0b1c5b622644ac17e505c9a049
+      env:
+        GITHUB_TOKEN: ${{ inputs.GITHUB_TOKEN }}
+        SONAR_TOKEN: ${{ inputs.SONAR_TOKEN }}
+      with:
+        args: >
+          -Dsonar.cpd.exclusions=frontend-react/**/*.test.*
+          -Dsonar.javascript.lcov.reportPaths=frontend-react/coverage/lcov.info
```

Comment:
my only concern here is that for reasons that I am not sure of, we seem to be not runnign tests when building for prod. In that case there won't be a coverage report for sonarcloud to read here. We could either:
* run tests in prod, which seems like it makes sense to me
* not run sonarCloud in prod, which if there is logic around not running tests in prod, it may make sense not to run sonarCloud either

Reply:
Ah, you are right we have a conditional in the action. I will add the yarn test:ci to the production step. Good catch!

Reply:
Hmm, this change will mean it reports both on master & production though - so the net will be that it saves one build (after creating the deployment branch) and will stop running for feature branches.